test/sqllogictest/tpch.slt

```diff
@@ -1707,7 +1712,9 @@ ORDER BY
 
 %10 =
 | Get %8 (l1)
-| ArrangeBy (#0, #7)
+| Reduce group=(#1, #0, #7)
```

Comment:
The partial pushdown in TPCH query 21 here is a performance regression. I did three trials of query 21 with 4 workers and logical compaction window 1ms.
before: 23540.196,21319.351,22349.341
after: 23120.997, 22810.060, 24878.105

The pushdown is not worth doing because `(#0)` is a key of %0 and thus a key of %4 and a key of %8. 

My code should avoid doing a partial pushdown when an input key is a subset of the group key of the proposed inner reduce. Upon investigation, the reason this partial pushdown is occurring is that the key information does not propagate up to %10. I'm not quite sure why. My best theory is that the key information for a Get(LocalId(something)) is not being properly updated as we transform the `RelationExpr`.